lib/src/client.rs

```diff
@@ -55,9 +55,7 @@ pub(crate) struct ClientInner {
 }
 
 impl ClientInner {
-    async fn from_config(
-        config: ClientConfig,
-    ) -> Result<(Self, Consensus, Option<Validator>), Error> {
+    async fn from_config(config: ClientConfig) -> Result<Client, Error> {
```

Comment:
I know this works, but I'm still not sure I like it semantically (i.e. a `ClientInner` method returning a `Client` which is supposed to be a wrapper around `ClientInner`). I think it would make sense to discuss this on the next core call because I'm also not sure why we're using this `Client`/`ClientInner` pattern here.